### ISSUE TITLE
""TypeError: 'NoneType' object is not iterable"" when instantiating keras.models.Model() or keras.models.Sequential()

### ISSUE THREAD

**Issue by KbeeArthur** [tensorflow/tensorflow#45578]
Whenever i try to instantiate a keras model either Model() or Sequential() i get **TypeError: 'NoneType' object is not iterable**. 
As an example I've taken screenshots of a Sequential model i wanted to create on the Mnist data using PyCharm as follows.

![capture-20201210-164722](https://user-images.githubusercontent.com/72258001/101787332-a4c33b00-3b07-11eb-8c4c-23ad30821723.png)

After running i got the following error message, #Note the first 3 lines are not important as they're just saying that there is no GPU found on my laptop;;; 

![capture-20201210-164800](https://user-images.githubusercontent.com/72258001/101788484-d38de100-3b08-11eb-9708-fd66b50d495d.png)
![capture-20201210-164815](https://user-images.githubusercontent.com/72258001/101788736-1ea7f400-3b09-11eb-8c8e-69d3b68ad83c.png)

I think my error is emanating from line 9 `network = models.Sequential()`.
I also think my imports are fine.  I have also tried running the same code in my jupyter notebook with the same apparent error. I also tried instantiating `model = models.Model()` from my notebook but with the same apparent error. 

The following packages are the ones mainly involved in deep learning and their respective versions on my laptop;
- python v3.7.7
- tensorflow 2.3.1
- keras 2.4.3

Please comment with anymore information you may need to assist me.
Thanks in advance.

**Comment by ymodak** [comment]
Please copy paste your code and the stacktrace in the issue template with use proper formatting.
Screenshots do not help in reproducing the problem. Thanks!

**Comment by KbeeArthur** [comment]
Okay thanks @ymodak let me copy paste my code and the stacktrace here.
And on this one I changed `keras` to `tf.keras` as you had previously advised

`import tensorflow as tf
from keras.datasets import mnist
from tensorflow.keras import layers
from keras import utils`

`(train_images, train_labels), (test_images, test_labels) = mnist.load_data()`

`network = tf.keras.Sequential()
network.add(layers.Dense(512, activation="relu", input_shape=(28 * 28,)))
network.add(layers.Dense(10, activation="softmax"))
network.compile(optimizer="rmsprop",
                loss="categorical_crossentropy",
                metrics=["accuracy"])`

`train_images = train_images.reshape((60000, 28 * 28))
train_images = train_images.astype("float32") / 255
test_images = test_images.reshape((10000, 28 * 28))
test_images = test_images.astype("float32") / 255`

`train_labels = utils.to_categorical(train_labels)
test_labels = utils.to_categorical(test_labels)
network.fit(train_images, train_labels, epochs=5, batch_size=128)
test_loss, test_acc = network.evaluate(test_images, test_labels)
print("test_acc:", test_acc)`

The output is as below on my notebook

`TypeError                                 Traceback (most recent call last)
<ipython-input-42-6cc7110698f3> in <module>
     5 (train_images, train_labels), (test_images, test_labels) = mnist.load_data()
      6 
----> 7 network = tf.keras.Sequential()
      8 network.add(layers.Dense(512, activation="relu", input_shape=(28 * 28,)))
      9 network.add(layers.Dense(10, activation="softmax"))`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\training\tracking\base.py in _method_wrapper(self, *args, **kwargs)
    455     self._self_setattr_tracking = False  # pylint: disable=protected-access
    456     try:
--> 457       result = method(self, *args, **kwargs)
    458     finally:
    459       self._self_setattr_tracking = previous_value  # pylint: disable=protected-access`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\keras\engine\sequential.py in __init__(self, layers, name)
    115     # Skip the init in FunctionalModel since model doesn't have input/output yet
    116     super(functional.Functional, self).__init__(  # pylint: disable=bad-super-call
--> 117         name=name, autocast=False)
    118     self.supports_masking = True
    119     self._compute_output_and_mask_jointly = True`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\training\tracking\base.py in _method_wrapper(self, *args, **kwargs)
    455     self._self_setattr_tracking = False  # pylint: disable=protected-access
    456     try:
--> 457       result = method(self, *args, **kwargs)
    458     finally:
    459       self._self_setattr_tracking = previous_value  # pylint: disable=protected-access`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\keras\engine\training.py in __init__(self, *args, **kwargs)
    306     self._steps_per_execution = None
    307 
--> 308     self._init_batch_counters()
    309     self._base_model_initialized = True
    310     _keras_api_gauge.get_cell('model').set(True)`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\training\tracking\base.py in _method_wrapper(self, *args, **kwargs)
    455     self._self_setattr_tracking = False  # pylint: disable=protected-access
    456     try:
--> 457       result = method(self, *args, **kwargs)
    458     finally:
    459       self._self_setattr_tracking = previous_value  # pylint: disable=protected-access`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\keras\engine\training.py in _init_batch_counters(self)
    315     # `evaluate`, and `predict`.
    316     agg = variables.VariableAggregationV2.ONLY_FIRST_REPLICA
--> 317     self._train_counter = variables.Variable(0, dtype='int64', aggregation=agg)
    318     self._test_counter = variables.Variable(0, dtype='int64', aggregation=agg)
    319     self._predict_counter = variables.Variable(`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\ops\variables.py in __call__(cls, *args, **kwargs)
    260       return cls._variable_v1_call(*args, **kwargs)
    261     elif cls is Variable:
--> 262       return cls._variable_v2_call(*args, **kwargs)
    263     else:
    264       return super(VariableMetaclass, cls).__call__(*args, **kwargs)`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\ops\variables.py in _variable_v2_call(cls, initial_value, trainable, validate_shape, caching_device, name, variable_def, dtype, import_scope, constraint, synchronization, aggregation, shape)
    254         synchronization=synchronization,
    255         aggregation=aggregation,
--> 256         shape=shape)
    257 
    258   def __call__(cls, *args, **kwargs):`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\ops\variables.py in <lambda>(**kws)
    235                         shape=None):
    236     """Call on Variable class. Useful to force the signature."""
--> 237     previous_getter = lambda **kws: default_variable_creator_v2(None, **kws)
    238     for _, getter in ops.get_default_graph()._variable_creator_stack:  # pylint: disable=protected-access
    239       previous_getter = _make_getter(getter, previous_getter)`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\ops\variable_scope.py in default_variable_creator_v2(next_creator, **kwargs)
   2644       synchronization=synchronization,
   2645       aggregation=aggregation,
-> 2646       shape=shape)
   2647 
   2648 `

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\ops\variables.py in __call__(cls, *args, **kwargs)
    262       return cls._variable_v2_call(*args, **kwargs)
    263     else:
--> 264       return super(VariableMetaclass, cls).__call__(*args, **kwargs)
    265 
    266 `

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\ops\resource_variable_ops.py in __init__(self, initial_value, trainable, collections, validate_shape, caching_device, name, dtype, variable_def, import_scope, constraint, distribute_strategy, synchronization, aggregation, shape)
   1516           aggregation=aggregation,
   1517           shape=shape,
-> 1518           distribute_strategy=distribute_strategy)
   1519 
   1520   def _init_from_args(self,`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\ops\resource_variable_ops.py in _init_from_args(self, initial_value, trainable, collections, caching_device, name, dtype, constraint, synchronization, aggregation, distribute_strategy, shape)
   1650             initial_value = ops.convert_to_tensor(
   1651                 initial_value() if init_from_fn else initial_value,
-> 1652                 name="initial_value", dtype=dtype)
   1653           if shape is not None:
   1654             if not initial_value.shape.is_compatible_with(shape):`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\framework\ops.py in convert_to_tensor(value, dtype, name, as_ref, preferred_dtype, dtype_hint, ctx, accepted_result_types)
   1497 
   1498     if ret is None:
-> 1499       ret = conversion_func(value, dtype=dtype, name=name, as_ref=as_ref)
   1500 
   1501     if ret is NotImplemented:`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\framework\tensor_conversion_registry.py in _default_conversion_function(***failed resolving arguments***)
     50 def _default_conversion_function(value, dtype, name, as_ref):
     51   del as_ref  # Unused.
---> 52   return constant_op.constant(value, dtype, name=name)
     53 
     54 `

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\framework\constant_op.py in constant(value, dtype, shape, name)
    262   """
    263   return _constant_impl(value, dtype, shape, name, verify_shape=False,
--> 264                         allow_broadcast=True)
    265 
    266 `

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\framework\constant_op.py in _constant_impl(value, dtype, shape, name, verify_shape, allow_broadcast)
    273       with trace.Trace("tf.constant"):
    274         return _constant_eager_impl(ctx, value, dtype, shape, verify_shape)
--> 275     return _constant_eager_impl(ctx, value, dtype, shape, verify_shape)
    276 
    277   g = ops.get_default_graph()`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\framework\constant_op.py in _constant_eager_impl(ctx, value, dtype, shape, verify_shape)
    298 def _constant_eager_impl(ctx, value, dtype, shape, verify_shape):
    299   """Implementation of eager constant."""
--> 300   t = convert_to_eager_tensor(value, ctx, dtype)
    301   if shape is None:
    302     return t`

`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\framework\constant_op.py in convert_to_eager_tensor(value, ctx, dtype)
     95     except AttributeError:
     96       dtype = dtypes.as_dtype(dtype).as_datatype_enum
---> 97   ctx.ensure_initialized()
     98   return ops.EagerTensor(value, ctx.device_name, dtype)
     99 
`
`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\eager\context.py in ensure_initialized(self)
    522       opts = pywrap_tfe.TFE_NewContextOptions()
    523       try:
--> 524         config_str = self.config.SerializeToString()
    525         pywrap_tfe.TFE_ContextOptionsSetConfig(opts, config_str)
    526         if self._device_policy is not None:
`
`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\eager\context.py in config(self)
    904     """Return the ConfigProto with all runtime deltas applied."""
    905     # Ensure physical devices have been discovered and config has been imported
--> 906     self._initialize_physical_devices()
    907 
    908     config = config_pb2.ConfigProto()
`
`c:\users\danda\appdata\local\programs\python\python37\lib\site-packages\tensorflow\python\eager\context.py in _initialize_physical_devices(self)
   1225       self._physical_devices = [
   1226           PhysicalDevice(name=d.decode(),
-> 1227                          device_type=d.decode().split(":")[1]) for d in devs]
   1228       self._physical_device_to_index = {
   1229           p: i for i, p in enumerate(self._physical_devices)
`
`TypeError: 'NoneType' object is not iterable`

**Comment by ymodak** [comment]
Can you try setting up a [virtual env](https://www.jetbrains.com/help/pycharm/creating-virtual-environment.html) and run following lines;
```python
import tensorflow as tf
from tensorflow.keras import layers
network = tf.keras.Sequential()
```

**Comment by KbeeArthur** [comment]
Unfortunately I got the same error even in a virtual environment

![capture-20201211-095104](https://user-images.githubusercontent.com/72258001/101877333-82c4c980-3b96-11eb-8132-cbbf3eeacf9a.png)

I thought the screenshot would be better to show my error rather than put in all the code like before since it's giving the exact same error.

**Comment by amahendrakar** [comment]
@KbeeArthur,
Could you please update TensorFlow to v2.4 and check if you are facing the same issue. Thanks!

**Comment by KbeeArthur** [comment]
@amahendrakar okay thanks let me do that and I'll get back to you just now. Thanks.

**Comment by KbeeArthur** [comment]
Thanks guys for all your assistance on the issue. After @amahendrakar suggested i download tensorflow 2.4, i initially downloaded using `pip install --upgrade tensorflow==2.4 --user` but the code still did not run but i got a new error on `network = tf.keras.models.Sequential()` which I'm sorry but did not take note of it at that time. I then went to the tensorflow [website](https://www.tensorflow.org/install/pip#system-install) where i got the package location and then used my cmd again and this time used pip to download but with location.

So it's just now that I've figured I must have initially downloaded a tensorflow version for gpu whilst my laptop does not have one. So I downloaded the **Python 3.7 CPU-only** version from the website and everything now works just fine.

Thanks! :100: :)
